src/client/app/effects/github.ts

```diff
@@ -109,9 +110,15 @@ export class GitHubEffects {
             }
 
             return this._github.createOrUpdateGist(
-                description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
+                    description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
             )
-            .map((gist: IGist) => ({ gist: gist, snippetId: id }));
+            .map((gist: IGist) => ({ gist: gist, snippetId: id }))
+            .catch(exception => {
+                if (exception.status === 404) {
+                    throw { id: id, gistOwnerId: UNKNOWN_GIST_OWNER_ID};
```

Comment:
You should always throw instances of an Error object (JS best practice).  If this is an intentional error, which this one is, that's OK, just use or create your own intentional/informational error class.  See "class InformationalError extends Error {" in the server code for an example.